boards: stm32l562e_dk: add support for touchscreen

lucbeaufils opened the pull request:
The stm32l562e_dk board uses a ft6x06 i2c controller for the touchscreen. The zephyr driver ft5336 can control it.

lucbeaufils committed:
boards: stm32l562e_dk: add support for touchscreen

The stm32l562e_dk board uses a ft6x06 i2c controller for the touchscreen.
The zephyr driver ft5336 can control it.

Signed-off-by: Luc BEAUFILS <luc.beaufils@savoirfairelinux.com>

erwango reviewed (changes requested):
Thanks for this addition. Would you mid updating board documentation ?
Otherwise LGTM.

lucbeaufils committed:
doc: stm32l562e_dk: add a screen and touch panel section

The screen and touch panel are working with zephyr. Add it in the board
documentation.

Signed-off-by: Luc BEAUFILS <luc.beaufils@savoirfairelinux.com>

lucbeaufils commented:
@erwango I add section for screen and touchpanel

erwango commented on boards/st/stm32l562e_dk/doc/index.rst:
```diff
@@ -280,6 +280,15 @@ Serial Port
 STM32L562E-DK Discovery board has 6 U(S)ARTs. The Zephyr console output is
 assigned to USART1. Default settings are 115200 8N1.
 
+TFT LCD screen and touch panel
+------------------------------
+
+The TFT LCD screen and touch panel are supported for the STM32L562E-DK Discovery board.
+They can be tested using :zephyr:code-sample:`lvgl` sample:
+
+.. code-block:: console
+
+   $ west build -b stm32l562e_dk samples/subsys/display/lvgl/
```
Actually I meant to update `Supported Features` section.

erwango reviewed (dismissed):
Thanks!

FRASTM commented on boards/st/stm32l562e_dk/doc/index.rst:
```diff
@@ -160,6 +160,9 @@ hardware features:
 +-----------+------------+-------------------------------------+
 | DAC       | on-chip    | DAC Controller                      |
 +-----------+------------+-------------------------------------+
+| DISPLAY   | on-chip    | TFT LCD screen with st7789v driver  |
```
The FMC peripheral which  controls the TFT -LCD driver is  **on-chip**. Does it actually make the display to be "on-chip" or "on-board"

kartben commented:
@lucbeaufils please rebase to resolve merge conflict

lucbeaufils commented:
> @lucbeaufils please rebase to resolve merge conflict

@kartben done

kartben commented on boards/st/stm32l562e_dk/doc/index.rst:
```diff
@@ -273,6 +276,15 @@ Serial Port
+The TFT LCD screen and touch panel are supported for the STM32L562E-DK Discovery board.
+They can be tested using :zephyr:code-sample:`lvgl` sample:
+
+.. code-block:: console
+
+   $ west build -b stm32l562e_dk samples/subsys/display/lvgl/
```
this should actually be using `.. zephyr-app-commands::` rather than hard-coding the build command. This also makes it easier to catch stale build instructions should e.g the sample be moved or whatever

```suggestion
.. zephyr-app-commands::
   :zephyr-app: samples/subsys/display/lvgl
   :board: stm32l562e_dk
   :goals: build
```

I should have also reviewed the PR more thoroughly when asking you to rebase, sorry to be asking you for another update

lucbeaufils replied:
I made the changes.
The other commands in this docs are also hardcoded and should be refactored.

kartben commented on boards/st/stm32l562e_dk/doc/index.rst:
```diff
@@ -273,6 +276,17 @@ Serial Port
+They can be tested using :zephyr:code-sample:`lvgl` sample:
+
+.. zephyr-app-commands::
+
```
sorry that sphinx is being uncooperative but adding that blank line is causing the directive to be interpreted different (I know this came from a good intention of trying to improve readbility :) )

lucbeaufils replied:
Good to know !  I don't know sphinx very well
